onert-micro/luci-interpreter/src/loader/GraphLoader.cpp

```diff
@@ -169,18 +169,15 @@ void GraphLoader::initInputTensors() const
     auto tensor_interpreter = std::make_unique<Tensor>(dtype, std::move(shape), quantization);
 
     tensor_interpreter->set_allocatable(_memory_manager->is_allocate_input());
-    if (not _memory_manager->is_static_manager())
+    if (not use_static_memory_manager)
     {
       // Using Dynamic Allocations
       _memory_manager->allocate_memory(*tensor_interpreter);
     }
     else
     {
       // Using static allocations
-      const auto static_memory_manager = dynamic_cast<StaticMemoryManager *>(_memory_manager);
-      if (static_memory_manager->is_owning_buffers())
-        // Using split buffers (input, output and for intermediate computations)
-        static_memory_manager->allocate_input_buf();
+      _memory_manager->allocate_input_buf();
```

Comment:
hmmm, maybe you can call both of these methods and do not pass `use_static_memory_manager` flag here?

Reply:
I'm afraid it will then be harder to read the code, so maybe let's return to this in the future?